### PR TITLE
Add CI for pulling charts from GH releases

### PR DESCRIPTION
**Pull request by indyfree**
Towards https://github.com/giantswarm/giantswarm/issues/4848.
Dependend on https://github.com/giantswarm/prometheus/pull/112.

Adds CI that publishes charts to the the hosted helm repostiory (https://giantswarm.github.io/app-catalog/). The chart tarballs are pulled from GH releases from apps that are specified in the `APP` file.

- The `ci-scripts/build-repo.sh` is based on the upstream CI script: https://github.com/helm/charts/blob/master/test/repo-sync.sh
(Do we need to include APACHE licencse here too then btw?)
- You can read up about the upstream helm repositories wer are using here: https://github.com/helm/helm/blob/master/docs/chart_repository.md
- TL;DR: The helm charts are hosted on this repo, which is published with GitHub Pages

Notes:
- This script-based solution is to have a first working app-catalog out there soon. Considerations about moving it to either GitHub Actions, CircleCI Orbs or integrate into our tooling (e.g. architect, devctl) should be taken in the future.
- **Caution:** The Circle Cronjob is set up to run every 15 Minutes. If multiple versions of one App are relased within the 15 minute interval, only the latest will get published. We should consider changing this in the future.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+workflows:
+  version: 2
+  lint:
+    jobs:
+      - lint
+  sync:
+    triggers:
+      - schedule:
+          cron: "10,25,40,55 * * * *"
```
Comment (rossf7) [inline .circleci/config.yml:28]: 👍 Every 15 mins makes sense to me.

File: ci-scripts/build-repository.sh [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+set -o errexit
+set -o nounset
+set -o pipefail
+
+readonly REPONAME=$1
+readonly PERSONAL_ACCESS_TOKEN=$2
+
+readonly APPS_FILE=./APPS
+readonly HELM_URL=https://storage.googleapis.com/kubernetes-helm
+readonly HELM_TARBALL=helm-v2.11.0-linux-amd64.tar.gz
```
Comment (rossf7) [inline ci-scripts/build-repository.sh]: Can we make this `2.12.0` as that matches `giantswarm/helmclient`.

**Review by rossf7 — APPROVED** [review]
LGTM just a minor thing to align the Helm version.

Thanks for taking care here. ❤️

File: ci-scripts/build-repository.sh (hunk 2 above)
Comment (indyfree) [reply ci-scripts/build-repository.sh]: Totally!

**Comment by sslavic** [comment]
Deprecated, closing since we switched from pull to push approach.
